docs/process/integrating-with-fxa.md

```diff
@@ -4,28 +4,123 @@ title: Integration with FxA
+8. [User data and account notifications are properly](#user-data-hygiene-gdpr) handled and GDPR compliant.
+9. An icon suitable to display in FxA’s [Devices & apps][devices-and-apps] list has been sent to FxA devs.
 
-## For Engineers
+### Preparing for Production
 
-Turnip greens yarrow ricebean rutabaga endive cauliflower sea lettuce kohlrabi amaranth water spinach avocado daikon napa cabbage asparagus winter purslane kale. Celery potato scallion desert raisin horseradish spinach carrot soko. Lotus root water spinach fennel kombu maize bamboo shoot green bean swiss chard seakale pumpkin onion chickpea gram corn pea. Brussels sprout coriander water chestnut gourd swiss chard wakame kohlrabi beetroot carrot watercress. Corn amaranth salsify bunya nuts nori azuki bean chick
+1. Register for a production OAuth credentials by filing a [deployment bug][deployment-bug]. See [OAuth credentials](#oauth-credentials).
+2. Production servers point to with https://oauth.accounts.firefox.com.
+3. Someone from the FxA team has reviewed the integration code and tested the flow.
 
-## For Designers
+### User Authentication with OAuth 2.0 / OpenID Connect in a nutshell
 
-Turnip greens yarrow ricebean rutabaga endive cauliflower sea lettuce kohlrabi amaranth water spinach avocado daikon napa cabbage asparagus winter purslane kale. Celery potato scallion desert raisin horseradish spinach carrot soko. Lotus root water spinach fennel kombu maize bamboo shoot green bean swiss chard seakale pumpkin onion chickpea gram corn pea. Brussels sprout coriander water chestnut gourd swiss chard wakame kohlrabi beetroot carrot watercress. Corn amaranth salsify bunya nuts nori azuki bean chick
+1. Create a state token, associate it with a local session.
+2. Send [/authentication request](#authorization-query-parameters) to Firefox Accounts. Upon completion, Firefox Accounts redirects back to your app with state and code.
+3. Confirm returned state token by comparing it with state token associated with the local session.
+4. Exchange code for an access token and possibly a refresh token (for clients that request offline use).
+5. Fetch user profile information with access token from the FxA Profile Server.
+6. Associate profile information with local session and possibly create an account.
+
+### OAuth Credentials
+
+1. client_id - a public identifier that is used to identify your service. Can be public.
+2. client_secret - a private secret that is sent from the backend when interacting with the OAuth server. Must not be shared publicly, checked into a public repository, or bundled with compiled code.
+
+### Self Hosted Email-first Flow
+
+1. Initialize top of funnel metrics by calling [/metrics-flow request][metrics-flow-request] with the required query parameters:
+   a. entrypoint
+   b. form_type (must be the string ‘email’)
+   c. utm_source
+   d. utm_campaign
+2. Propagate email, flow_id and flow_begin_time query parameters to /authentication request
```

Comment:
it may not be obvious that the flow parameters are returned from the above call.